### PR TITLE
feat: add `LinearRegression.predict_explain()` to generate `ML.EXPLAIN_PREDICT`  columns

### PR DESCRIPTION
**Pull request by arwas11**
Thank you for opening a Pull Request! Before submitting your PR, there are a few things you can do to make sure it goes smoothly:
- [ ] Make sure to open an issue as a [bug/issue](https://github.com/googleapis/python-bigquery-dataframes/issues/new/choose) before writing your code!  That way we can discuss the change, evaluate designs, and agree on the general idea
- [ ] Ensure the tests and linter pass
- [ ] Code coverage does not decrease (if any source code was changed)
- [x] Appropriate docs were updated (if necessary)
  - `predict_Explain`: screen/BmpXWdueDEz9K9P

Fixes #379743762 🦕

### CODE REVIEW

File: tests/system/small/ml/test_linear_model.py [hunk 1]
```diff
@@ -106,6 +106,26 @@ def test_linear_reg_model_predict(penguins_linear_model, new_penguins_df):
     )
 
 
+def test_linear_reg_model_predict_explain(penguins_linear_model, new_penguins_df):
+    predictions = penguins_linear_model.predict_explain(new_penguins_df).to_pandas()
+    assert predictions.shape == (3, 12)
+    result = predictions[["predicted_body_mass_g", "approximation_error"]]
```
Comment (GarrettWu) [inline tests/system/small/ml/test_linear_model.py:112]: We may want to check the existence of other columns. https://cloud.google.com/bigquery/docs/reference/standard-sql/bigqueryml-syntax-explain-predict

And maybe some values if they give meaningful results. (such as probability, top_feature_attributions) https://cloud.google.com/bigquery/docs/reference/standard-sql/bigqueryml-syntax-explain-predict#output

File: bigframes/ml/sql.py [hunk 2]
```diff
@@ -304,6 +304,11 @@ def ml_predict(self, source_sql: str) -> str:
         return f"""SELECT * FROM ML.PREDICT(MODEL {self._model_ref_sql()},
   ({source_sql}))"""
 
+    def ml_explain_predict(self, source_sql: str) -> str:
```
Comment (GarrettWu) [inline bigframes/ml/sql.py:307]: add a unit test to make sure we don't accidentally break it.

https://github.com/googleapis/python-bigquery-dataframes/blob/main/tests/unit/ml/test_sql.py

File: bigframes/ml/linear_model.py [hunk 3]
```diff
@@ -160,6 +160,14 @@ def predict(self, X: utils.ArrayType) -> bpd.DataFrame:
 
         return self._bqml_model.predict(X)
 
+    def predict_explain(self, X: utils.ArrayType) -> bpd.DataFrame:
```
Comment (GarrettWu) [inline bigframes/ml/linear_model.py:163]: Need to add docs to the functions.

Comment (tswast) [inline bigframes/ml/linear_model.py:163]: Looks like there are a few parameters available, too. https://cloud.google.com/bigquery/docs/reference/standard-sql/bigqueryml-syntax-explain-predict

Consider adding support for these. For example, I believe `top_k_features` is relevant for the linear model and is used in the example for it. https://cloud.google.com/bigquery/docs/reference/standard-sql/bigqueryml-syntax-explain-predict#explain_a_prediction_generated_by_a_linear_regression_model

If not, at least add a TODO for these parameters.

File: tests/system/small/ml/test_linear_model.py (hunk 1 above)
Comment (tswast) [reply tests/system/small/ml/test_linear_model.py:112]: +1 to `top_feature_attributions` as that's the most important part of `ML.EXPLAIN_PREDICT` from my understanding of the function.

Even if we don't check for the exact values (which might be a bit brittle), checking that these columns are present is important. See: https://github.com/googleapis/python-bigquery-dataframes/blob/33928008f64ddc4ae88432bcff6daa7ec64e807c/tests/system/small/ml/test_forecasting.py#L135-L159 for a test that checks if the expected columns are there.

File: bigframes/ml/sql.py (hunk 2 above)
Comment (tswast) [reply bigframes/ml/sql.py:307]: https://github.com/googleapis/python-bigquery-dataframes/blob/33928008f64ddc4ae88432bcff6daa7ec64e807c/tests/unit/ml/test_sql.py#L333-L343 for an example.

**Review by tswast — APPROVED** [review]

**Comment by tswast** [comment]
Doctest failed, but doesn't appear to be related:

```
FAILED third_party/bigframes_vendored/pandas/core/frame.py::frame.DataFrame.update
```

```
_______________________ [doctest] frame.DataFrame.update _______________________
[gw1] linux -- Python 3.12.0 /tmpfs/src/github/python-bigquery-dataframes/.nox/doctest/bin/python
4066 
4067             >>> import bigframes.pandas as bpd
4068             >>> bpd.options.display.progress_bar = None
4069 
4070             >>> df = bpd.DataFrame({'A': [1, 2, 3],
4071             ...                    'B': [400, 500, 600]})
4072             >>> new_df = bpd.DataFrame({'B': [4, 5, 6],
4073             ...                        'C': [7, 8, 9]})
4074             >>> df.update(new_df)
4075             >>> df
UNEXPECTED EXCEPTION: BadRequest('GET [https://bigquery.googleapis.com/bigquery/v2/projects/bigframes-testing/queries/a0e298c3-d88a-43a3-8ee8-06454d818cbe?maxResults=0&location=US&prettyPrint=false](https://www.google.com/url?q=https://bigquery.googleapis.com/bigquery/v2/projects/bigframes-testing/queries/a0e298c3-d88a-43a3-8ee8-06454d818cbe?maxResults%3D0%26location%3DUS%26prettyPrint%3Dfalse&sa=D): The job encountered an internal error during execution and was unable to complete successfully.')
```

**Pull request merged**